halo2_gadgets/benches/poseidon.rs

```diff
@@ -221,30 +169,40 @@ fn bench_poseidon<S, const WIDTH: usize, const RATE: usize, const L: usize>(
 
     let circuit = HashCircuit::<S, WIDTH, RATE, L> {
         message: Value::known(message),
-        output: Value::known(output),
         _spec: PhantomData,
     };
 
+    // Create a proof
+    let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
+
     c.bench_function(&prover_name, |b| {
         b.iter(|| {
-            // Create a proof
-            let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
-            create_proof(&params, &pk, &[circuit], &[&[]], &mut rng, &mut transcript)
-                .expect("proof generation should not fail")
+            create_proof(
+                &params,
+                &pk,
+                &[circuit],
+                &[&[&[output]]],
+                &mut rng,
+                &mut transcript,
+            )
+            .expect("proof generation should not fail")
         })
     });
 
-    // Create a proof
-    let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
-    create_proof(&params, &pk, &[circuit], &[&[]], &mut rng, &mut transcript)
-        .expect("proof generation should not fail");
     let proof = transcript.finalize();
```

Comment:
This causes the created proof to be a concatenation of all of the proofs created in the previous benchmark. It happens to work because the proof benchmark creates valid proofs, and `verify_proof` does not check that the transcript is empty at the end, so it will just repeatedly consume the first proof in `proof`. But it's not an intended benchmark pattern, and means that we hold a (relatively) large allocation in memory. Undo this change.

Reply:
Have reverted these changes